src/unicode/utf8/utf8_spec.gobra

```diff
@@ -0,0 +1,70 @@
+package utf8
+//+gobra
+
+import (
+	. "verification/utils/definitions"
+	sl "gobra-libs/byteslice"
+)
+
+// utf8.Codepoints(s) returns the sequence of (utf-8 encoded) codepoints in `s`.
+// Handles invalid utf-8 by replacing offending bytes with RuneError.
+//
+// See utf8.DecodeRune
+ghost
+pure
+ensures 0 < len(s) ==> 0 < len(res)
+ensures 0 == len(s) ==> 0 == len(res)
+trusted
+decreases
+func Codepoints(s []byte) (res seq[rune])
+
+
+const (
+	RuneError = '\uFFFD'
+	RuneSelf  = 0x80
+	MaxRune   = '\U0010FFFF'
+	UTFMax    = 4
+)
+
+// AppendRune appends the UTF-8 encoding of r to the end of p and
+// returns the extended buffer. If the rune is out of range,
+// it appends the encoding of [RuneError].
+func AppendRune(p []byte, r rune) []byte
```

Comment:
I would expect that this function requires full permission to `p` and returns full permission to the result